src/form-field/__tests__/form-field-analytics.test.tsx

```diff
@@ -0,0 +1,159 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import FormField from '../../../lib/components/form-field';
+
+import { FunnelMetrics, setFunnelMetrics } from '../../../lib/components/internal/analytics';
+
+import {
+  AnalyticsFunnel,
+  AnalyticsFunnelStep,
+  AnalyticsFunnelSubStep,
+} from '../../../lib/components/internal/analytics/components/analytics-funnel';
+
+const mockedFunnelInteractionId = 'mocked-funnel-id';
+function mockFunnelMetrics() {
+  setFunnelMetrics({
+    funnelStart: jest.fn(() => mockedFunnelInteractionId),
+    funnelError: jest.fn(),
+    funnelComplete: jest.fn(),
+    funnelSuccessful: jest.fn(),
+    funnelCancelled: jest.fn(),
+    funnelStepStart: jest.fn(),
+    funnelStepComplete: jest.fn(),
+    funnelStepNavigation: jest.fn(),
+    funnelSubStepStart: jest.fn(),
+    funnelSubStepComplete: jest.fn(),
+    funnelSubStepError: jest.fn(),
+    helpPanelInteracted: jest.fn(),
+    externalLinkInteracted: jest.fn(),
+  });
+}
+
+describe('FormField Analytics', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockFunnelMetrics();
+  });
+
+  test('sends funnelSubStepError metric when errorText is present', () => {
+    render(
+      <AnalyticsFunnel funnelType="single-page" optionalStepNumbers={[]} totalFunnelSteps={1}>
+        <AnalyticsFunnelStep stepNumber={2} stepNameSelector=".step-name-selector">
+          <AnalyticsFunnelSubStep>
+            <FormField label="Label" errorText="Error" />
+          </AnalyticsFunnelSubStep>
+        </AnalyticsFunnelStep>
+      </AnalyticsFunnel>
+    );
+
+    expect(FunnelMetrics.funnelSubStepError).toHaveBeenCalledTimes(1);
+    expect(FunnelMetrics.funnelSubStepError).toHaveBeenCalledWith(
+      expect.objectContaining({
+        funnelInteractionId: 'mocked-funnel-id',
+        fieldErrorSelector: expect.any(String),
+        fieldLabelSelector: expect.any(String),
+        subStepSelector: expect.any(String),
+        stepNameSelector: expect.any(String),
+        subStepAllSelector: expect.any(String),
+        subStepNameSelector: expect.any(String),
```

Comment:
Would it make sense to check for any more specific strings?

Reply:
Most of the selectors use the useUniqueId hook so they have generated strings in them. Is it a good trade off to mock the module in this test?

Reply:
If it's not too cumbersome, that would be great. Yeah

Reply:
I can maybe do as a follow up. I got push back in a different PR where I mocked a module.